configuration/configuration.go

```diff
@@ -834,6 +867,54 @@ func (c *Client) setFieldValue(section parser.Section, sectionName string, field
 		}
 		return nil
 	}
+	if fieldName == "ExternalCheck" {
+		enabled := field.Bool()
+		if section == parser.Global {
+			if enabled {
+				d := &types.Enabled{}
+				if err := p.Set(section, sectionName, "external-check", d); err != nil {
+					return err
+				}
+			} else {
+				_ = p.Delete(section, sectionName, "external-check")
+			}
+		} else if section == parser.Backends || section == parser.Defaults {
+			d := &types.SimpleOption{
+				NoOption: !enabled,
+			}
+			if err := p.Set(section, sectionName, "option external-check", d); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	if fieldName == "ExternalCheckPath" {
+		if valueIsNil(field) {
+			return nil
+		}
+		extPath := field.Interface().(string)
+		d := &types.ExternalCheckPath{
+			Path: extPath,
+		}
+		if err := p.Set(section, sectionName, "external-check path", d); err != nil {
+			return err
+		}
+		return nil
+	}
+	if fieldName == "ExternalCheckCommand" {
+		if valueIsNil(field) {
+			return nil
```

Comment:
Also if it is nil, you have to delete this row, set it to nil

Reply:
I think "set it to nil" is unnecessary.

Reply:
See https://github.com/haproxytech/client-native/pull/6/files#r299485939

Reply:
Done.

Reply:
```
if valueIsNil(field) {
    if err := p.Set(section, sectionName, "external-check path", nil); err != nil {
	return err
    }
    return nil
}
```

Reply:
I believe following piece is equivalent to your suggestion:
```
	if fieldName == "ExternalCheckCommand" {
		pExtCmd := &types.ExternalCheckCommand{}
		if valueIsNil(field) {
			pExtCmd = nil
		} else {
			pExtCmd.Command = field.Interface().(string)
		}
		if err := p.Set(section, sectionName, "external-check command", pExtCmd); err != nil {
			return err
		}
		return nil
	}

```